packages/base-contract/src/index.ts

```diff
@@ -46,20 +45,20 @@ export class BaseContract {
         // 2. Global config passed in at library instantiation
         // 3. Gas estimate calculation + safety margin
         const removeUndefinedProperties = _.pickBy;
-        const txDataWithDefaults = {
+        const txDataWithDefaults = ({
             to: this.address,
             ...removeUndefinedProperties(this._web3Wrapper.getContractDefaults()),
             ...removeUndefinedProperties(txData as any),
             // HACK: TS can't prove that T is spreadable.
             // Awaiting https://github.com/Microsoft/TypeScript/pull/13288 to be merged
-        };
+        } as any) as TxData;
```

Comment:
Why not: `const txDataWithDefaults: TxData = ...`?

Reply:
Because this type is not directly convertible to TxData.
In line 51 we convert txData to any and loose the info that from is defined. If I try to directly assign - it complaints that from might be undefined. So I first convert to any and then to txData.